core/lib/zksync_core/src/api_server/web3/namespaces/eth.rs

```diff
@@ -816,16 +828,26 @@ impl<G: L1GasPriceProvider> EthNamespace<G> {
                 } else {
                     vec![]
                 };
+
+                let mut to_block = self
+                    .state
+                    .resolve_filter_block_number(filter.to_block)
+                    .await?;
+
+                if let Some(BlockNumber::Number(_)) = filter.to_block {
```

Comment:
Nit: You can use `if matches!(filter.to_block, Some(BlockNumber::Number(_)))`, since you don't bind any vars in the matched pattern.